### PR TITLE
IDMP-592 - Complete the preliminary modeling work required for Nucleic Acid, covering all detail provided in ISO 11238

### PR DESCRIPTION
**Pull request by ElisaKendall**
Description: 
1. Added metadata to nucleic acid substance, added nucleic acid sequence type per the ISO specification
2. Added nucleic acid subunit and related properties and restrictions on both nucleic acid and subunit
3. Added sugar and linkage, and built up the relevant attributes to complete the representation

### CODE REVIEW

**Review by mereolog — CHANGES_REQUESTED** [review]

File: ISO/ISO11238-Substances.rdf [hunk 1]
```diff
@@ -5555,6 +5921,43 @@ In all these cases, a new ID shall be issued, that is, &apos;Human Albumin, Plas
...
+		<skos:definition>relates a nucleic acid subunit to the unique identifier of the five prime</skos:definition>
+		<skos:example>JKUHT76541 (Artificial ID)</skos:example>
+		<idmp-sub:hasConformanceLevel rdf:resource="&idmp-sub;ConformanceLevel-Conditional"/>
+		<idmp-sub:hasValuesAllowed>ISO 11238 Substance ID</idmp-sub:hasValuesAllowed>
+		<idmp-dtp:isEncodedAs rdf:resource="&idmp-dtp;DatatypeCode-II"/>
+		<cmns-av:adaptedFrom>ISO/TS 19844:2018(E) Health informatics - Identification of medicinal products (IDMP) - Implementation guidelines for ISO 11238 for data elements and structures for the unique identification and exchange of regulated information on substances, clause and 7.3 and Figure 27</cmns-av:adaptedFrom>
+		<cmns-av:directSource>ISO 11238:2018 Health informatics - Identification of medicinal products (IDMP) - Data elements and structures for the unique identification and exchange of regulated information on substances, clause 7.5</cmns-av:directSource>
+	</owl:ObjectProperty>
+	
+	<owl:ObjectProperty rdf:about="&idmp-sub;hasFivePrimeSubstanceName">
```
Comment (tw-osthus) [inline ISO/ISO11238-Substances.rdf:5946]: Why do we need "has related substance name"/"has reltated substance identifier" and their subproperties, when it is enough to have "has related substance/has substance name". If we want to refer from the nucleic acid to the sugar, a property "has sugar" as subclass of has related substance or has moiety make more sense to me. 
Referencing to related entities via ids or names is relational modeling, not graph based modeling.
The 3/5 prime and sugar are moieties or constituents of the nucleic acid.

File: ISO/ISO11238-Substances.rdf [hunk 2]
```diff
@@ -5690,6 +6093,54 @@ In all these cases, a new ID shall be issued, that is, &apos;Human Albumin, Plas
 		<cmns-av:adaptedFrom>ISO/TS 19844:2018(E) Health informatics - Identification of medicinal products (IDMP) - Implementation guidelines for ISO 11238 for data elements and structures for the unique identification and exchange of regulated information on substances, clause 6.7.7.3</cmns-av:adaptedFrom>
 	</owl:ObjectProperty>
 	
+	<owl:DatatypeProperty rdf:about="&idmp-sub;hasLinkageConnectivity">
```
Comment (tw-osthus) [inline ISO/ISO11238-Substances.rdf:6096]: as this expects a code it should be an object property pointing to a controlled term. Then their is no need to encode the allowed values as "3’→5’, 3’→3’ or 5’→5’" which is human readable, but not machine readable.

File: ISO/ISO11238-Substances.rdf [hunk 3]
```diff
@@ -6224,6 +6706,19 @@ In all these cases, a new ID shall be issued, that is, &apos;Human Albumin, Plas
 		<cmns-av:explanatoryNote>It may be known that lysine residues are modified but the particular lysine in the sequence may not be known. It should be captured for both specific and non-specific modifications, e.g. sizing carbohydrate polymers by hydrolysis.</cmns-av:explanatoryNote>
 	</owl:ObjectProperty>
 	
+	<owl:DatatypeProperty rdf:about="&idmp-sub;hasResidueSite">
```
Comment (tw-osthus) [inline ISO/ISO11238-Substances.rdf:6709]: Again this should be an object property refering to a residue site, which is a locator in the structure. It is fine to start with a description of the locator, but if we have a machine readable encoding of residue site, we will have to define another distinct property "has residue site as object". This is not what we want, and we should keep the pattern using property paths for such cases, when we can expect a machine readable way of exactly specifying the location.

**Review by tw-osthus — CHANGES_REQUESTED** [review]

File: ISO/ISO11238-Substances.rdf [hunk 4]
```diff
@@ -6249,6 +6744,32 @@ In all these cases, a new ID shall be issued, that is, &apos;Human Albumin, Plas
...
+		<rdfs:range rdf:resource="&cmns-doc;Reference"/>
+		<skos:definition>provides an &apos;enriched&apos; string, a document or image representing the sequence</skos:definition>
+		<idmp-sub:hasBusinessRules>The element Sequence is always mandatory; the element Sequence Attachment may be provided based on jurisdictional guidance. Either the sequence is provided as ST or/and as an attachment as ED.</idmp-sub:hasBusinessRules>
+		<idmp-sub:hasConformanceLevel rdf:resource="&idmp-sub;ConformanceLevel-Conditional"/>
+		<idmp-dtp:isEncodedAs rdf:resource="&idmp-dtp;DatatypeCode-ED"/>
+		<cmns-av:adaptedFrom>ISO/TS 19844:2018(E) Health informatics - Identification of medicinal products (IDMP) - Implementation guidelines for ISO 11238 for data elements and structures for the unique identification and exchange of regulated information on substances, clause 7.3 and Figure 27</cmns-av:adaptedFrom>
+		<cmns-av:directSource>ISO 11238:2018 Health informatics - Identification of medicinal products (IDMP) - Data elements and structures for the unique identification and exchange of regulated information on substances, clause 7.5</cmns-av:directSource>
+	</owl:DatatypeProperty>
```
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply ISO/ISO11238-Substances.rdf]: @merelog Done

File: ISO/ISO11238-Substances.rdf (hunk 1 above)
Comment (ElisaKendall) [reply ISO/ISO11238-Substances.rdf:5946]: @tw-osthus I thought about that, but there are specific queries that may be easier with explicit properties, and the standard requires it. Thus, we've used property chains, to connect through the substance to its name or identifier here.

File: ISO/ISO11238-Substances.rdf (hunk 2 above)
Comment (ElisaKendall) [reply ISO/ISO11238-Substances.rdf:6096]: @tw-osthus I did not find controlled vocabularies for this, but didn't search SPOR. The standard indicates that it is a string value, for this property.

File: ISO/ISO11238-Substances.rdf (hunk 3 above)
Comment (ElisaKendall) [reply ISO/ISO11238-Substances.rdf:6709]: @tw-osthus The goal this round was to represent what is in the standard. I'd like to defer these changes to review work once we have the initial approach done, and then refine this based on understanding the approach that the SMEs want to take and whether or not there are controlled vocabularies for this. Again, I didn't search, but think we should review with SMEs before making this change.

**Review by avilesg — APPROVED** [review]

**Review by mereolog — APPROVED** [review]

**Pull request merged**